test/unit/mdc-shape/foundation.test.js

```diff
@@ -0,0 +1,234 @@
+  td.verify(mockAdapter.setClipPathData('m 0 0 h 88 v 188'));
+});
+
+test('#setBackground changes fillStyle when redrawing', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.getCanvasWidth()).thenReturn(100);
+  td.when(mockAdapter.getCanvasHeight()).thenReturn(200);
+  td.when(mockAdapter.getDevicePixelRatio()).thenReturn(2);
+  const mockContext = td.object(MDCShapeFoundation.defaultAdapter.create2dRenderingContext());
+  td.when(mockAdapter.create2dRenderingContext()).thenReturn(mockContext);
+  const fillStyles = [];
+  mockContext.fill = () => {
+    fillStyles.push(mockContext.fillStyle);
+  };
+
+  foundation.init();
+  foundation.setBackground('#FF0000');
+  foundation.redraw();
+
+  assert.equal(fillStyles[3], '#FF0000');
+});
+
+test('#setElevation changes shadows when redrawing', () => {
+  const {foundation, mockAdapter} = setupTest();
+  foundation.generatePath_ = (width, height, padding) => {
+    return 'm ' + padding + ' ' + padding + ' h ' + width + ' v ' + height;
+  };
+  td.when(mockAdapter.getCanvasWidth()).thenReturn(100);
+  td.when(mockAdapter.getCanvasHeight()).thenReturn(200);
+  td.when(mockAdapter.getDevicePixelRatio()).thenReturn(2);
+  const mockContext = td.object(MDCShapeFoundation.defaultAdapter.create2dRenderingContext());
+  td.when(mockAdapter.create2dRenderingContext()).thenReturn(mockContext);
+  const shadowBlurs = [];
+  const shadowOffsetYs = [];
+  mockContext.fill = () => {
+    shadowBlurs.push(mockContext.shadowBlur);
+    shadowOffsetYs.push(mockContext.shadowOffsetY);
+  };
+
+  foundation.init();
+  foundation.setElevation(2, false);
+  foundation.redraw();
+
+  assert.equal(shadowBlurs[0], 10);
+  assert.equal(shadowBlurs[1], 4);
+  assert.equal(shadowBlurs[2], 2);
+  assert.equal(shadowOffsetYs[0], 2);
+  assert.equal(shadowOffsetYs[1], 4);
+  assert.equal(shadowOffsetYs[2], 6);
+  td.verify(mockAdapter.createPath2D('m 58 58 h 100 v 200'));
+});
+
+testFoundation('#setElevation changes shadows when animating', ({foundation, mockAdapter, mockRaf}) => {
+  foundation.generatePath_ = (width, height, padding) => {
+    return 'm ' + padding + ' ' + padding + ' h ' + width + ' v ' + height;
+  };
+  td.when(mockAdapter.getCanvasWidth()).thenReturn(100);
+  td.when(mockAdapter.getCanvasHeight()).thenReturn(200);
+  td.when(mockAdapter.getDevicePixelRatio()).thenReturn(2);
+  const mockContext = td.object(MDCShapeFoundation.defaultAdapter.create2dRenderingContext());
+  td.when(mockAdapter.create2dRenderingContext()).thenReturn(mockContext);
+  let fillCount = 0;
+  const shadowBlurs = [];
+  const shadowOffsetYs = [];
+  mockContext.fill = () => {
+    fillCount++;
+    shadowBlurs.push(mockContext.shadowBlur);
+    shadowOffsetYs.push(mockContext.shadowOffsetY);
+  };
+
+  foundation.init();
+  foundation.setElevation(2, true);
+  mockRaf.flush();
+
+  assert.equal(fillCount, 4);
+  assert.equal(shadowBlurs[0], 1);
+  assert.equal(shadowBlurs[1], 0.4);
```

Comment:
Should some of these values (either here or below after flushing) be looked up from the constants in mdc-elevation?